### PR TITLE
Await Vercel deployment to be ready before running E2E

### LINKED ISSUE #184: Sometimes, E2E tests fail because they're testing the Vercel domain instead of the deployment domain
It happens from time to time that Vercel didn't update the DNS of the newly deployed domain name, and Cypress runs too early and gets redirected to the deployment page, which is protected against unauthenticated users and gets redirected to the homepage.

The deployment page automatically redirects to the deployment domain once the domain has been published, but the login page doesn't.

Therefore, Cypress stays on that login page and perform the tests against the wrong domain, by testing Vercel auth page instead of the deployment.
Using the Cypress `attempts` feature doesn't solve the problem, because Cypress stays on the login page.

I'm not sure what can be done here, maybe cypress should keep redirecting to the deployment url until it works?  

---

### Resources:
- https://glebbahmutov.com/blog/develop-preview-test/

image

### PR DESCRIPTION
**Pull request by Demmonius**
https://app.asana.com/0/1136912812619441/1198898897367222

Use our recently released GitHub action https://github.com/UnlyEd/github-action-await-vercel to wait for the Vercel deployment to reach `status=READY` before running E2E/LH.

See https://github.com/UnlyEd/next-right-now/issues/184 for more details.

Fixes #184
Closes #184

### CODE REVIEW

File: .github/workflows/deploy-vercel-staging.yml
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: Awaiting Vercel deployment to be ready

File: .github/workflows/deploy-vercel-staging.yml
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: The deployment is

**Review by Vadorequest — CHANGES_REQUESTED** [review]
Looks great!

**Comment by Demmonius** [comment]
Fixed !
Before any merge, I will have to put the same code in the production file.
And we may change the version if you update the action. So let's wait for the review of the action ? :)

**Comment by Vadorequest** [comment]
Agreed, let's first focus on the Action itself, then merge this PR. But it looks good already :)

**Comment by Demmonius** [comment]
@Vadorequest if [our action](https://github.com/UnlyEd/github-action-await-vercel/) is fine for you, I think we can merge this. I just updated for the last release :)

File: .github/workflows/deploy-vercel-staging.yml
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: oula le sale copier coller. 😅

File: .github/workflows/deploy-vercel-staging.yml
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml:179]: We're gonna have a lot of conflicts with https://github.com/UnlyEd/next-right-now/pull/225, I'd rather merge https://github.com/UnlyEd/next-right-now/pull/225 first and adapt this PR then, because https://github.com/UnlyEd/next-right-now/pull/225 is more complicated to debug.

**Review by Vadorequest — CHANGES_REQUESTED** [review]

**Pull request merged**
